app/models/miq_widget.rb

```diff
@@ -458,7 +458,7 @@ def self.get_group(group)
   end
 
   def self.sync_from_dir
-    Dir.glob(File.join(WIDGET_DIR, "*.yaml")).sort.each { |f| sync_from_file(f) }
```

Comment:
It looks like it is also being used here https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_widget.rb#L559-L564, although it looks like that function is only being used in spec tests?

Reply:
Yeah you're right, I wonder if that could be replaced with `before { MiqWidget.seed }` in the specs?

Reply:
I think the idea behind that method was a way to seed a specific widget to avoid seeding _all_ of the widgets, mainly for test timing reasons.  We'd have to look at the callers and see how useful that is or not.

Reply:
Even so, I think we need to fix that caller, technically, before merging this.  I did something similar in other places by having a method that returns all of the possible directories, then using _that_ in the various sync methods. Here is an example with MiqReport: https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_report/seeding.rb#L91-L103

Reply:
@agrare `before { MiqWidget.seed }` doesn't seem to replace the behaviour.  Could we replicate something like `before { MiqWidget.sync_from_file(File.join(WIDGET_DIR, "*yaml_file*")) }` in the specs instead? Effectively pulling the `#self.seed_widget` function to the specs since `WIDGET_DIR` is only used there

Reply:
That's just a suggestion...if you want to drop seed_widget that's fine too - it's only meant as a helper method for testing and setting up a small number of widgets

Reply:
I've made changes based on my comment above and our discussion that we can remove `#seed_widget` since there are only a few callers